File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int s=0;
        int e=0;
        while(e<nums.length){
            if(nums[s]==0 && nums[e]!=0){
                int temp=nums[s];
                nums[s]=nums[e];
                nums[e]=temp;
            }
            else if(nums[s]==0 && nums[e]==0){
                e++;
            }
            else{
                s++;
                e++;
            }
        }
    }
}

File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int i=-1;

        for(int j=0; j<nums.size(); ++j){
          if(nums[j] != 0 && i!=-1){
            swap(nums[j], nums[i]);
            ++i;
          }
          else if(nums[j] == 0 && i == -1){
            i = j;
          }
        }
    }
};